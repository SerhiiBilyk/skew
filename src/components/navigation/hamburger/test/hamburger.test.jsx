import React from 'react';
import Navigation from '../../navigation.jsx';
import Hamburger from '../hamburger.jsx';
import {shallow, mount} from 'enzyme';

import renderer from 'react-test-renderer';

import {HashRouter as Router, Route, Link, Switch} from 'react-router-dom';



/**SNAPSHOT*/
it('<Navigation/> snapshot', () => {
  const tree = renderer.create(
    <Router><Navigation/></Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<Hamburger/>', () => {
  it('onclick', () => {
    const wrapper = mount(
      <Router><Navigation/></Router>
    );
    expect(wrapper.find('#hamburger').hasClass('expanded')).toEqual(false)
    wrapper.find('#hamburger').simulate('click')
    expect(wrapper.find('#hamburger').hasClass('expanded')).toEqual(true)
  });
});
