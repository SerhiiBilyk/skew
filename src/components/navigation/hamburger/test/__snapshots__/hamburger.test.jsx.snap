// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> snapshot 1`] = `
<div
  className="navigation"
>
  <div
    className="wrapper"
  >
    <div
      className="logo"
    >
       
      <img
        src="test-file-stub"
      />
    </div>
    <div
      className="bar"
    >
      <div
        className="wrapper"
      >
        <span
          className="height-fix"
        >
           
        </span>
        ;
        <ul
          className="hamburger"
          id="hamburger"
          onClick={[Function]}
        >
          <li />
          <li />
          <li />
        </ul>
      </div>
    </div>
  </div>
  <div
    className="scrollbar"
  >
    <ul
      className="items mobile"
      deep={0}
    >
      <li
        className="item"
        deep={undefined}
        onClick={[Function]}
        state={1.1}
      >
        
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            Home
          </a>
        </span>
      </li>
      <li
        className="item"
        deep={undefined}
        onClick={[Function]}
        state={1.1}
      >
        
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            About
          </a>
        </span>
      </li>
      <li
        className="container item"
        deep={0}
        onClick={[Function]}
        state={1.1}
      >
        <span
          className="wrapper down"
        >
          <span
            className="fa fa-angle-up"
          />
        </span>
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            Portfolio
          </a>
        </span>
        <ul
          className="sub"
          deep={1}
        >
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                first
              </a>
            </span>
          </li>
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                first
              </a>
            </span>
          </li>
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                first
              </a>
            </span>
          </li>
          <li
            className="container item"
            deep={1}
            onClick={[Function]}
            state={1.1}
          >
            <span
              className="wrapper down"
            >
              <span
                className="fa fa-angle-up"
              />
            </span>
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                first
              </a>
            </span>
            <ul
              className="sub"
              deep={2}
            >
              <li
                className="item"
                deep={undefined}
                onClick={[Function]}
                state={1.1}
              >
                
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    second
                  </a>
                </span>
              </li>
              <li
                className="container item"
                deep={2}
                onClick={[Function]}
                state={1.1}
              >
                <span
                  className="wrapper down"
                >
                  <span
                    className="fa fa-angle-up"
                  />
                </span>
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    second
                  </a>
                </span>
                <ul
                  className="sub"
                  deep={3}
                >
                  <li
                    className="item"
                    deep={undefined}
                    onClick={[Function]}
                    state={1.1}
                  >
                    
                    <span
                      className="link"
                    >
                      <a
                        href="#/home"
                        onClick={[Function]}
                      >
                        third
                      </a>
                    </span>
                  </li>
                  <li
                    className="container item"
                    deep={3}
                    onClick={[Function]}
                    state={1.1}
                  >
                    <span
                      className="wrapper down"
                    >
                      <span
                        className="fa fa-angle-up"
                      />
                    </span>
                    <span
                      className="link"
                    >
                      <a
                        href="#/home"
                        onClick={[Function]}
                      >
                        third
                      </a>
                    </span>
                    <ul
                      className="sub"
                      deep={4}
                    >
                      <li
                        className="item"
                        deep={undefined}
                        onClick={[Function]}
                        state={1.1}
                      >
                        
                        <span
                          className="link"
                        >
                          <a
                            href="#/home"
                            onClick={[Function]}
                          >
                            fourth
                          </a>
                        </span>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li
        className="item"
        deep={undefined}
        onClick={[Function]}
        state={1.1}
      >
        
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            Blog
          </a>
        </span>
      </li>
      <li
        className="item"
        deep={undefined}
        onClick={[Function]}
        state={1.1}
      >
        
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            Contacts
          </a>
        </span>
      </li>
      <li
        className="container item"
        deep={0}
        onClick={[Function]}
        state={1.1}
      >
        <span
          className="wrapper down"
        >
          <span
            className="fa fa-angle-up"
          />
        </span>
        <span
          className="link"
        >
          <a
            href="#/home"
            onClick={[Function]}
          >
            Features
          </a>
        </span>
        <ul
          className="sub"
          deep={1}
        >
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                Multipage
              </a>
            </span>
          </li>
          <li
            className="container item"
            deep={1}
            onClick={[Function]}
            state={1.1}
          >
            <span
              className="wrapper down"
            >
              <span
                className="fa fa-angle-up"
              />
            </span>
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                Options
              </a>
            </span>
            <ul
              className="sub"
              deep={2}
            >
              <li
                className="container item"
                deep={2}
                onClick={[Function]}
                state={1.1}
              >
                <span
                  className="wrapper down"
                >
                  <span
                    className="fa fa-angle-up"
                  />
                </span>
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    General
                  </a>
                </span>
                <ul
                  className="sub"
                  deep={3}
                >
                  <li
                    className="item"
                    deep={undefined}
                    onClick={[Function]}
                    state={1.1}
                  >
                    
                    <span
                      className="link"
                    >
                      <a
                        href="#/home"
                        onClick={[Function]}
                      >
                        Kapitan
                      </a>
                    </span>
                  </li>
                </ul>
              </li>
              <li
                className="item"
                deep={undefined}
                onClick={[Function]}
                state={1.1}
              >
                
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    Sidebars
                  </a>
                </span>
              </li>
              <li
                className="item"
                deep={undefined}
                onClick={[Function]}
                state={1.1}
              >
                
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    Fonts
                  </a>
                </span>
              </li>
              <li
                className="item"
                deep={undefined}
                onClick={[Function]}
                state={1.1}
              >
                
                <span
                  className="link"
                >
                  <a
                    href="#/home"
                    onClick={[Function]}
                  >
                    Socials
                  </a>
                </span>
              </li>
            </ul>
          </li>
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                Page
              </a>
            </span>
          </li>
          <li
            className="item"
            deep={undefined}
            onClick={[Function]}
            state={1.1}
          >
            
            <span
              className="link"
            >
              <a
                href="#/home"
                onClick={[Function]}
              >
                FAQ
              </a>
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
